### PR TITLE
Fix opinionated block styles loading in editor.

### PR DESCRIPTION
**Pull request by jffng**
## What?

This PR checks if the current theme has declared support for `wp-block-styles` before adding the dependency in the site editor.

## Why?

Right now the opinionated styles are being loaded in the editor, regardless if the theme has declared support. 

## How?

I traced the problem to the 6.0 compat file.

## Testing Instructions

1. Activate `emptytheme`
2. Go to the site editor, verify theme.css is loaded. A quick visual check is adding a separator at the root of the page and seeing that the width is limited to 100px.
3. Comment out / remove the `add_theme_support( 'wp-block-styles' );` line of the theme's functions.php  
4. Refresh the site editor, verify the theme.css is no longer loaded. (The separator's width should not be limited.)

## Screenshots or screencast 

cc @WordPress/block-themers

### CODE REVIEW

**Review by mikachan — COMMENTED** [review]
This is working well for me following the test instructions. I can see that theme.css is not loaded in the editor unless the theme has opted into `wp-block-styles`.

File: lib/compat/wordpress-6.0/client-assets.php [hunk 1]
```diff
@@ -45,10 +45,13 @@ function gutenberg_resolve_assets() {
...
 		'wp-block-editor',
 		'wp-block-library',
-		'wp-block-library-theme',
 		'wp-edit-blocks',
 	);
 
+	if ( current_theme_supports( 'wp-block-styles' ) ) {
+		$style_handles[] = 'wp-block-library-theme';
+	}
+
```
Comment (draganescu) [inline lib/compat/wordpress-6.0/client-assets.php:54]: How is this related to the similar code in `gutenberg_edit_site_init`

```
if (
		current_theme_supports( 'wp-block-styles' ) ||
		( ! is_array( $editor_styles ) || count( $editor_styles ) === 0 )
	) {
		wp_enqueue_style( 'wp-block-library-theme' );
	}
```

Comment (jffng) [reply lib/compat/wordpress-6.0/client-assets.php:54]: I am not sure, but I did try changing those conditions (removing the enqueue) and it did not affect whether the theme.css was loaded in the site editor.

Comment (scruffian) [reply lib/compat/wordpress-6.0/client-assets.php:54]: I think its duplicated because this is a compat file, but it introduces this bug into 6.0. IMO this fix should be backported to 6.0.

**Review by scruffian — APPROVED** [review]

File: lib/compat/wordpress-6.0/client-assets.php (hunk 1 above)
Comment (scruffian) [reply lib/compat/wordpress-6.0/client-assets.php:54]: cc @adamziel @gziolo

Comment (gziolo) [reply lib/compat/wordpress-6.0/client-assets.php:54]: Added to the 6.0 board in the Triage column. @ndiego will take care of everything else as usual. Thank you.

**Comment by ndiego** [comment]
I am struggling to replicate the original issue. I am using Gutenberg 13.1 and 6.0 RC1. I have also tested with just RC1. With `emptytheme`, the separator is 100px. When I comment out that `add_theme_support( 'wp-block-styles' );`, it goes full width. This is the intended functionality, correct?

**Comment by scruffian** [comment]
> This is the intended functionality, correct?

Correct

**Comment by ndiego** [comment]
@scruffian, but my testing above was **without** this PR applied.

**Comment by scruffian** [comment]
> @scruffian, but my testing above was without this PR applied.

Did you test in the Site Editor, not the Post Editor? That's the only environment where I see the issue.

**Comment by ndiego** [comment]
> Did you test in the Site Editor, not the Post Editor? That's the only environment where I see the issue.

🤦‍♂️ My apologies, I was looking in the Post Editor. Yes, I can replicate it in the Site Editor and this PR fixes the issue.

**Comment by ndiego** [comment]
@gziolo @adamziel this is a pretty important one as it introduces unexpected behavior for themes that have disabled block styles. I recommend we backport this once merged.

**Review by aristath — APPROVED** [review]

**Comment by jffng** [comment]
Thanks everyone for the reviews and @scruffian for the rebase.

**Comment by adamziel** [comment]
I went ahead and cherry-picked this PR to the wp/6.0 branch to get it included in the next release: b99494292aea422f7a053363507e1da2d9d8b78c

**Comment by gziolo** [comment]
This change has to be applied manually in WordPress core.

**Comment by gziolo** [comment]
Committed to WP Core with https://github.com/WordPress/wordpress-develop/commit/af8844c28767dddd211a792a1e05ca43da8ba2ac.
